### PR TITLE
ETX-534: Java Codegen - `ContractDecoder.fromEvent` can lookup via package name

### PR DESCRIPTION
**Pull request by raphael-speyer-da**
This allows the codegen to decode contracts from newer packages

### CODE REVIEW

File: sdk/canton/community/bindings-java/src/main/java/com/daml/ledger/javaapi/data/codegen/ContractDecoder.java [hunk 1]
```diff
@@ -21,18 +22,40 @@ public ContractDecoder(
...
           companions) {
     this.companions =
-        StreamSupport.stream(companions.spliterator(), false)
-            .collect(Collectors.toMap(c -> c.TEMPLATE_ID, Function.identity()));
+        Stream.concat(
+                // Each companion should be keyed by TEMPLATE_ID and by TEMPLATE_ID_WITH_PACKAGE_ID
+                StreamSupport.stream(companions.spliterator(), false)
+                    .map(c -> Map.entry(c.TEMPLATE_ID, c)),
+                StreamSupport.stream(companions.spliterator(), false)
+                    .map(c -> Map.entry(c.TEMPLATE_ID_WITH_PACKAGE_ID, c)))
```
Comment (erwin-ramirez-da) [inline sdk/canton/community/bindings-java/src/main/java/com/daml/ledger/javaapi/data/codegen/ContractDecoder.java:30]: we can use a single stream and avoid the concat
``` 
   StreamSupport.stream(companions.spliterator(), false)
       .flatMap(c -> Stream.of(
            Map.entry(c.TEMPLATE_ID, c),
            Map.entry(c.TEMPLATE_ID_WITH_PACKAGE_ID, c)
       ))
```

**Review by tudor-da — APPROVED** [review]
lgtm!

**Review by erwin-ramirez-da — APPROVED** [review]
apart from the small change suggestion LGTM

File: sdk/canton/community/bindings-java/src/main/java/com/daml/ledger/javaapi/data/codegen/ContractDecoder.java (hunk 1 above)
Comment (raphael-speyer-da) [reply sdk/canton/community/bindings-java/src/main/java/com/daml/ledger/javaapi/data/codegen/ContractDecoder.java]: Done

**Pull request merged**